### PR TITLE
Correct the HMAC calculation using VSE password

### LINKED ISSUE #1578: Incorrect HMAC computation from VSE password for synthetic scripts
### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_synthetics_monitor_script`

### Actual Behavior
The HMAC computed from the VSE password in 2.34.0 and 2.34.1 is incorrect and causes all HMACs using the new `vse_password` field in the location block to be incorrect. This means any synthetic using VSE passwords will fail to run in locations requiring VSE.

### Expected Behavior
Using the `vse_password` field should result in correct HMACs

### Important Factoids
As can be seen from the steps in the [documentation](https://newrelic.zendesk.com/hc/en-us/articles/1500005170762-Calculating-the-HMAC-value-for-a-VSE-Password-on-your-private-location) for computing the HMAC, the HMAC must provided as the base64 encoding of the _hexadecimal encoded_ SHA256 sum. The algorithm used in the provider skips the hexadecimal encoding step.

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- #1436
- #1533

### PR DESCRIPTION
**Pull request by zeffron**
According to the steps in the [documentation](https://newrelic.zendesk.com/hc/en-us/articles/1500005170762-Calculating-the-HMAC-value-for-a-VSE-Password-on-your-private-location) for computing the HMAC for use with Synthetic scripts, the algorithm for computing the HMAC requires base64 encoding the _hexadecimal encoded_ HMAC. The previously implemented algorithm skipped the hexadecimal encoding resulting in incorrect values.

Fixes #1578

### CODE REVIEW

**Comment by mbazhlekova** [comment]
@zeffron Thanks for the contribution! All looks good to me. Just one thing, can you sign the CLA?

**Comment by zeffron** [comment]
> @zeffron Thanks for the contribution! All looks good to me. Just one thing, can you sign the CLA?

Just waiting on corporate approval to do so. Hopefully I'll have that approval today.

**Comment by zeffron** [comment]
Got the approval and CLA is signed.

**Review by mbazhlekova — APPROVED** [review]

**Pull request merged**
